File: backend/src/main/java/thkoeln/dungeon/player/domain/NameGenerator.java
package thkoeln.dungeon.player.domain;

import java.util.Random;

/**
 * (c) https://jvm-gaming.org/t/super-simple-name-generator/53855
 */
public class NameGenerator {
    private static final String[] Beginning = {"Kr", "Ca", "Ra", "Mrok", "Cru",
            "Ray", "Bre", "Zed", "Drak", "Mor", "Jag", "Mer", "Jar", "Mjol",
            "Zork", "Mad", "Cry", "Zur", "Creo", "Azak", "Azur", "Rei", "Cro",
            "Mar", "Luk"};
    private static final String[] Middle = {"air", "ir", "mi", "sor", "mee", "clo",
            "red", "cra", "ark", "arc", "miri", "lori", "cres", "mur", "zer",
            "marac", "zoir", "slamar", "salmar", "urak"};
    private static final String[] End = {"d", "ed", "ark", "arc", "es", "er", "der",
            "tron", "med", "ure", "zur", "cred", "mur"};

    private static final Random rand = new Random();

    public static String generateName() {

        return Beginning[rand.nextInt(Beginning.length)] +
                Middle[rand.nextInt(Middle.length)] +
                End[rand.nextInt(End.length)] +
                rand.nextInt(1000);
    }
}

File: src/main/java/thkoeln/dungeon/player/domain/NameGenerator.kt
package thkoeln.dungeon.player.domain

import java.util.*

/**
 * (c) https://jvm-gaming.org/t/super-simple-name-generator/53855
 */
object NameGenerator {
    private val Beginning = arrayOf(
        "Kr", "Ca", "Ra", "Mrok", "Cru",
        "Ray", "Bre", "Zed", "Drak", "Mor", "Jag", "Mer", "Jar", "Mjol",
        "Zork", "Mad", "Cry", "Zur", "Creo", "Azak", "Azur", "Rei", "Cro",
        "Mar", "Luk"
    )
    private val Middle = arrayOf(
        "air", "ir", "mi", "sor", "mee", "clo",
        "red", "cra", "ark", "arc", "miri", "lori", "cres", "mur", "zer",
        "marac", "zoir", "slamar", "salmar", "urak"
    )
    private val End = arrayOf(
        "d", "ed", "ark", "arc", "es", "er", "der",
        "tron", "med", "ure", "zur", "cred", "mur"
    )
    private val rand = Random()
    fun generateName(): String {
        return Beginning[rand.nextInt(
            Beginning.size
        )] +
                Middle[rand.nextInt(Middle.size)] +
                End[rand.nextInt(End.size)] + rand.nextInt(1000).toString()
    }
}